app/controllers/active_storage/blobs_controller.rb

```diff
@@ -0,0 +1,39 @@
+# NOTE TO READER: this replacement of ActiveStorage's blob controller
+# is in place to enforce authentication when accessing a blob. At the
+# time of writing, the `/rails/active_storage/blobs/XXX` path is not
+# protected which means the Rails app will happily generate a valid
+# link to the underlying service+asset and redirect anyone who asks
+# there.
+#
+# However ActiveStorage will naturally evolve and probably implement a
+# way to do this without redefining the controller itself but in the
+# meantime it means we've started diverging. This should be kept in
+# sight and ultimately removed.
+#
+# PS: a more direct `before_action: authenticate_user!` was causing an
+# error, probably because the filter is meant to be used in the
+# context of a Rails app rather than its internal engines.
+
+# frozen_string_literal: true
+
+# Take a signed permanent reference for a blob and turn it into an expiring service URL for download.
+# Note: These URLs are publicly accessible. If you need to enforce access protection beyond the
+# security-through-obscurity factor of the signed blob references, you'll need to implement your own
+# authenticated redirection controller.
+class ActiveStorage::BlobsController < ActiveStorage::BaseController
+  include ActiveStorage::SetBlob
+  include Rails.application.routes.url_helpers
+
+  before_action :check_user
+
+  def check_user
+    if current_user.nil?
+      redirect_to new_user_session_path
+    end
+  end
+
+  def show
```

Comment:
this does not solve the issue that anyone can see an organisation MOU, as long as they have an admin account. I can see 3 ways to solve this:

1. add a, say, `/mou/signed_mou/` route path to the mou controller that bypasses and does the redirect itself (bit hacky)
2. on show, test `current_organisation.signed_mou.blob`equals `@blob` similar to what you are doing in https://github.com/alphagov/govwifi-admin/commit/e131f7073cb38d604421dd7416e96f84da0fc229
3. or you can do 2. but on `check_user` instead

I think I have a preference for 2.

Reply:
Yeah that makes sense. I thought we could get away with just making sure the MOUs are only accessed from within the app but I'll scope it to the correct organisation as well.

Reply:
I don't think 1 is is feasible but I might be confused.

Reply:
yeah, my initial impression was that 1. would be too hacky and moves the concern outside the context. I prefer 2. (or 3.)